### DISCUSSION TITLE
`NEXT_PUBLIC_BASE_PATH` not work with docker

### DISCUSSION THREAD

**Discussion by YealKio** [lobehub/lobehub#2528] [Self Hosting]
### 💻 Operating System

Ubuntu

### 📦 Environment

Docker

### 🌐 Browser

Chrome

### 🐛 Bug Description

Chat-lobe + ollama +nginx not work.
Chat-lobe can not init in the first time, like nginx SUBPATH thing, not sure, see my step please.

BTW, ollama server and Chat-lobe server are not the same device, but in LAN.
I can run OPEN-WEBUI + OLLAMA + NGINX without problem.

### 🚦 Expected Behavior

Even can not init.

### 📷 Recurrence Steps

Chat lobe run with docker: 
`sudo docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=http://192.168.2.202:11434 -e NEXT_PUBLIC_BASE_PATH=/chat-lobe  lobehub/lobe-chat`
this my nginx conf file:

```

upstream chat-lobe {
   server 127.0.0.1:3210;
}

    location  /chat-lobe/ {
         proxy_pass http://chat-lobe/;
         proxy_set_header Host $host;
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Proto $scheme;
         access_log /var/log/nginx/chat-lobe.access.log;
         error_log /var/log/nginx/chat-lobe.error.log;
    }
```

when I open my.nas.com:port/chat-lobe, the Chrome just show:
![image](https://github.com/lobehub/lobe-chat/assets/48641500/52bbef73-127f-4cbf-a030-1e411096c8fc)

The website console like:

![image](https://github.com/lobehub/lobe-chat/assets/48641500/e367ff62-8807-478c-a587-28aa965e7ce0)

Can not find any thing, and  one of Request URL  like: `Request URL:
http://my.nas.com:port/_next/static/chunks/webpack-72cd2e677540060b.js` 

It look like NEXT_PUBLIC_BASE_PATH  not work as expected, please help!

### 📝 Additional Information

Ollama server and Chat-lobe server are not the same device, but in LAN.

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
NEXT_PUBLIC_BASE_PATH is not working with docker. You need to build it with `NEXT_PUBLIC_` env by yourself

**Reply by YealKio** [reply]
Got it, so I have to modify the Dockerfile and build it local right?
Can you please show me a docker build cmd in this project? Hope your reply, thanks!

**Reply by arvinxx** [reply]
> Got it, so I have to modify the Dockerfile and build it local right?

no need to modify the Dockerfile, we have add the `NEXT_PUBLIC_` env in the Dockerfile , just add a `.env` file then run the docker build is fine.

**Reply by YealKio** [reply]
Hi @arvinxx ,thanks for your tips. I try to build docker image like following:
I create a `.env` file in porject root path and write `NEXT_PUBLIC_BASE_PATH="/chat-lobe"`,
then I start build image with ` sudo docker build -t lobe-chat .` and finally success.
Of cause I run docker `sudo docker run -d -p 3210:3210 -e OLLAMA_PROXY_URL=http://192.168.2.202:11434 -e NEXT_PUBLIC_BASE_PATH=/chat-lobe  lobe-chat` success.
But it looks NEXT_PUBLIC_BASE_PATH still not in work. 
Did I missing something?

**Reply by arvinxx** [reply]
> But it looks NEXT_PUBLIC_BASE_PATH still not in work.

what's the problem you meet?

**Reply by YealKio** [reply]
same as 
![image](https://github.com/lobehub/lobe-chat/assets/48641500/05313835-4cf6-491d-ac4c-e281e6d40660)

nginx conf
```
upstream chat-lobe {
   server 127.0.0.1:3210;
}

    location  /chat-lobe/ {
         proxy_pass http://chat-lobe/;
         proxy_set_header Host $host;
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Proto $scheme;
         access_log /var/log/nginx/chat-lobe.access.log;
         error_log /var/log/nginx/chat-lobe.error.log;
    }
```

Browser can not load noarmlly.

**Reply by arvinxx** [reply]
@YealKio  do you try 

```
upstream chat-lobe {
   server 127.0.0.1:3210;
}

    location / {
         proxy_pass http://chat-lobe/;
         proxy_set_header Host $host;
         proxy_set_header X-Real-IP $remote_addr;
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Proto $scheme;
         access_log /var/log/nginx/chat-lobe.access.log;
         error_log /var/log/nginx/chat-lobe.error.log;
    }
```

and then visit `/chat-lobe` ?

**Reply by YealKio** [reply]
Hi @arvinxx ! Actually it work just visit `http://myweb:port/`, I use offical docker image. So that's why I say `NEXT_PUBLIC_BASE_PATH` didn't work in above issues.https://github.com/lobehub/lobe-chat/discussions/2528#discussioncomment-9454328

**Reply by YealKio** [reply]
The question is how can I visit `chat-lobe` in a web sub-path. Chat-lobe is a amazing thing and the only obstacles between us is the `NEXT_PUBLIC_BASE_PATH`, so any other messages can I upload to help us deal with it, I am willing to do.

**Reply by arvinxx** [reply]
@YealKio After you build with `docker docker build` , do you just try with `ip:port/basePath` ? It should make sure it's not the issue with nginx

**Comment by LouisCan** [comment]
This is a bug; the parameter isn't taking effect.

**Reply by arvinxx** [reply]
it's hard to maintain this feature. I will remove it soon, please don't use this parameter
